cli/src/main.rs

```diff
@@ -53,15 +53,35 @@ fn main() -> Result<(), Error> {
                 .help("Determines who is winning. Positive number indicates a white advantage.")
                 .action(clap::ArgAction::SetTrue),
         )
+        .arg(
+            Arg::new("move")
+                .long("move")
+                .short('m')
+                .value_name("move")
+                .help("Make a move from starting square to final square (ex: 'E2:E4'"),
+        )
         .get_matches();
-
+    // Happen every time //
     let fen = matches
         .get_one::<String>("fen")
         .ok_or(Error::ArgumentError("FEN string required".to_string()))?;
-    let gs = GameState::from_string(fen).map_err(|e| Error::FENParsingError(e.to_string()))?;
 
-    // Setters //
+    let mut gs = GameState::from_string(fen).map_err(|e| Error::FENParsingError(e.to_string()))?;
 
+    // Setters //
+    if let Some(move_coords) = matches.get_one::<String>("move") {
+        let mut move_coords = move_coords.split(':');
+        if let Some(start_square) = move_coords.next() {
+            let from = position::string_to_square(start_square)
+                .map_err(|e| Error::ArgumentError(e.to_string()))?;
+            if let Some(end_square) = move_coords.next() {
+                let to = position::string_to_square(end_square)
+                    .map_err(|e| Error::ArgumentError(e.to_string()))?;
+                gs.move_piece_legally(from, to)
+                    .map_err(|e| Error::ArgumentError(e.to_string()))?;
+            }
+        }
+    }
```

Comment:
There has to be a cleaner way to write this.

Reply:
Bet